packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -40,6 +61,167 @@ abstract class BindingBase {
     assert(() { _debugInitialized = true; return true; });
   }
 
+  /// Called, in checked mode, to register service extensions.
```

Comment:
That sounds fine. Or the driver extension could have a separate registration path. I imagine most of the time developers will have separate entry points for their driver test apps that use fake server backends and such (for example the test app for Stocks could replace fetching data from the Internet with a canned list of stock tickers). For production builds we should probably disable observatory altogether, so registering service extensions is a noop (or an error).